### PR TITLE
2024q1 release

### PR DESCRIPTION
**Pull request by salkinium**
- [x] Update submodules
  - [x] CMSIS Headers for STM32
  - ~~LVGL v9.1.0~~ not motivated to fix it now
- [x] Release log
- [x] Changelog update

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -52,6 +52,170 @@ pay attention to. Medium impact changes are also worth looking at.
...
+- Support for ADC conversion sequences on STM32F3 and STM32G4.
+- Add Prescaler+Counter constexpr algorithm and use it in `Iwdg::initialize()`.
+- Add PWM modes, break input and comparator to Timer driver on STM32.
+- CMake generator now works with multiple lbuild repositories.
+- Allow setting I2C interrupt priority for STM32 and SAMx7x.
+
+Integrated Projects:
+
+- TinyUSB upgraded to v0.16.0.
+- LVGL upgraded to v9.1.0.
```
Comment (salkinium) [inline CHANGELOG.md]: nope

File: docs/release/2024q1.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+- TinyUSB v0.16 implements a DFU class in addition to a DFU *runtime* class.
+  Therefore the `modm:tinyusb:config:device.dfu` option was renamed to
+  `device.dfu_rt` to align with the TinyUSB naming.
+- TinyUSB v0.16 improves support for using two USB ports at the same time. A
+  differnt port can now be chosen for device and host classes. In addition, the
+  HS peripheral may be operated in FS mode, therefore the `modm:tinyusb:speed`
+  option was renamed to `modm:tinyusb:max-speed` since it now has a different
+  meaning.
+- Remove the separate `hosted-rpi` Raspberry Pi target in favor of plain
+  `hosted-linux`. Its GPIO implementation based on wiringPi is deprecated.
```
Comment (rleh) [inline docs/release/2024q1.md]:
```suggestion
  `hosted-linux`. Its GPIO implementation based on wiringPi is removed.
```

It was completely removed: https://github.com/modm-io/modm/pull/1113/files#diff-4e051bf200c2151a5c5c01a48684cbf46891af9ca6f477ebad753c28d806ebc0

File: docs/release/2024q1.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+#### 2024-02-01: Add BDMA driver for STM32H7
+
+PR #1115 -> 2e40ab4.  
+Tested in hardware by @chris-durand.
+
+#### 2024-01-21: Implement Fibers for ARM64
+
+PR #1113 -> 623a13b.  
+Tested in hardware by @salkinium.
```
Comment (rleh) [inline docs/release/2024q1.md]:
```suggestion
#### 2024-01-21: Implement Fibers for ARM64

- New targets with `-arm64` suffix, `-x86_64` is implicit for backwards compatibility:
  - `hosted-linux-arm64`
  - `hosted-darwin-arm64`
- Removed `hosted-rpi` target, as wiringPi is deprecated and thus the target lost its only peripheral.

PR #1113 -> 623a13b.  
Tested in hardware by @salkinium (Apple Silicon M2) and @rleh (Raspberry Pi).
```

File: docs/release/2024q1.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+- Niklas Hauser (@salkinium)
+- Raphael Lehmann (@rleh)
+- Thomas Sommer (@TomSaw)
+- Victor Costa (@victorandrehc)
+
+PR #1147 -> 2024q1.
+
+<details>
+<summary>Detailed changelog</summary>
+
```
Comment (rleh) [inline docs/release/2024q1.md:96]:
```suggestion

#### 2024-03-15: Allow I2C interrupt priority to be set

PR #1143 -> 734de07 and PR #1148 -> e1d8a17.  

```

File: docs/release/2024q1.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+#### 2024-01-05: Add DISCO-F401VC board support
+
+PR #1117 -> 2381c61.  
+Tested in hardware by @cajt.
+
+#### 2024-01-02: Miscellaneous DMA fixes for STM32H7
+
+PR #1114 -> 8bcbe25.  
+Tested in hardware by @chris-durand.
```
Comment (rleh) [inline docs/release/2024q1.md]:
```suggestion
#### 2024-01-04: Miscellaneous fixes for STM32H7 and STM32G0

PR #1114 -> 8bcbe25.  
Tested in hardware by @chris-durand.
```

File: docs/release/2024q1.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+PR #1147 -> 2024q1.
+
+<details>
+<summary>Detailed changelog</summary>
+
+#### 2024-03-09: Adapt CMake generator to multiple lbuild repositories
+
+PR #1140 -> 1a89fbe.  
+Tested in hardware by @Javask.
+
```
Comment (rleh) [inline docs/release/2024q1.md:105]:
```suggestion

#### 2024-03-08: Extend Timer features for STM32G0

PR #1134 -> a371df6.

```

File: docs/release/2024q1.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  HS peripheral may be operated in FS mode, therefore the `modm:tinyusb:speed`
+  option was renamed to `modm:tinyusb:max-speed` since it now has a different
+  meaning.
+- Remove the separate `hosted-rpi` Raspberry Pi target in favor of plain
+  `hosted-linux`. Its GPIO implementation based on wiringPi is deprecated.
+
+Features:
+
+- Add USB High-Speed support to all development boards with ULPI interface.
+- Catch accidental use of Newlib's printf implementation.
```
Comment (rleh) [inline docs/release/2024q1.md:22]: Feature or fix?

**Review by rleh — COMMENTED** [review]
Thank you!

File: docs/release/2024q1.md (hunk 7 above)
Comment (salkinium) [reply docs/release/2024q1.md:22]: The detection of this problem is a new feature, there is no fix other than to include `modm:printf`, but this needs to be decided by the user. So it's just better error reporting basically.

**Comment by salkinium** [comment]
Thanks, incorporated all your review, will merge when the CI passed, latest tomorrow morning.

**Pull request merged**